katharsis-client/src/main/java/io/katharsis/client/KatharsisClient.java

```diff
@@ -0,0 +1,226 @@
+	}
+
+	@SuppressWarnings({ "rawtypes", "unchecked" })
+	private <T, ID extends Serializable> void allocateRepository(Class<T> resourceClass) {
+		ResourceInformation resourceInformation = moduleRegistry.getResourceInformationBuilder().build(resourceClass);
+		final ResourceRepositoryStub<T, ID> repositoryStub = new ResourceRepositoryStubImpl<T, ID>(this, resourceClass,
+				resourceInformation, urlBuilder);
+
+		// create interface for it!
+		RepositoryInstanceBuilder repositoryInstanceBuilder = new RepositoryInstanceBuilder(null, null) {
+			@Override
+			public Object buildRepository() {
+				return repositoryStub;
+			}
+		};
+		ResourceEntry<T, ID> resourceEntry = new DirectResponseResourceEntry<T, ID>(repositoryInstanceBuilder);
+		Set<ResourceField> relationshipFields = resourceInformation.getRelationshipFields();
+		List<ResponseRelationshipEntry<T, ?>> relationshipEntries = new ArrayList<ResponseRelationshipEntry<T, ?>>();
+		RegistryEntry<T> registryEntry = new RegistryEntry<T>(resourceInformation, resourceEntry, relationshipEntries);
+
+		for (ResourceField relationshipField : relationshipFields) {
+			final Class<?> targetClass = relationshipField.getType();
+			final RelationshipRepositoryStubImpl relationshipRepositoryStub = new RelationshipRepositoryStubImpl(this,
+					resourceClass, targetClass, resourceInformation, urlBuilder, registryEntry);
+			RepositoryInstanceBuilder<RelationshipRepository> relationshipRepositoryInstanceBuilder = new RepositoryInstanceBuilder<RelationshipRepository>(
+					null, null) {
+				@Override
+				public RelationshipRepository buildRepository() {
+					return relationshipRepositoryStub;
+				}
+			};
+			DirectResponseRelationshipEntry relationshipEntry = new DirectResponseRelationshipEntry(
+					relationshipRepositoryInstanceBuilder) {
+
+				@Override
+				public Class<?> getTargetAffiliation() {
+					return targetClass;
+				}
+			};
+			relationshipEntries.add(relationshipEntry);
+		}
+		resourceRegistry.addEntry(resourceClass, registryEntry);
+	}
+
+	/**
+	 * @param resourceClass
+	 * @return stub for the given resourceClass
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public <T, ID extends Serializable> ResourceRepositoryStub<T, ID> getRepository(Class<T> resourceClass) {
+		init();
+
+		RegistryEntry<T> entry = resourceRegistry.getEntry(resourceClass);
+
+		// TODO fix this in katharsis, should be able to get original resource
+		ResourceRepositoryAdapter repositoryAdapter = entry.getResourceRepository(null);
+		return (ResourceRepositoryStub<T, ID>) repositoryAdapter.getResourceRepository();
+	}
+
+	/**
+	 * @param sourceClass
+	 * @param targetClass
+	 * @return stub for the relationship between the given source and target
+	 *         class
+	 */
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	public <T, ID extends Serializable, D, D_ID extends Serializable> RelationshipRepositoryStub<T, ID, D, D_ID> getRepository(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00119)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00119)